### PR TITLE
LPS-97485

### PR DESCRIPTION
**Pull request by yxingwu**
Hi @topolik ,

This PR just use ModelPermissionsFactory to fix the existing places in portal again as we talk about before (https://github.com/topolik/liferay-portal/pull/794#issuecomment-514326344).
About SF rules, I've talked with @hhuijser and create a ticket [LPS-99876](https://issues.liferay.com/browse/LPS-99876)

Regards.

### CODE REVIEW

**Comment by yxingwu** [comment]
ci:test:sf

**Comment by yxingwu** [comment]
ci:test:relevant

File: modules/apps/sync/sync-service/src/main/java/com/liferay/sync/internal/util/SyncHelperImpl.java [hunk 1]
```diff
@@ -518,7 +519,7 @@ public void setFilePermissions(
 			serviceContext.getModelPermissions();
 
 		if (modelPermissions == null) {
-			modelPermissions = new ModelPermissions();
+			modelPermissions = ModelPermissionsFactory.create();
```
Comment (topolik) [inline modules/apps/sync/sync-service/src/main/java/com/liferay/sync/internal/util/SyncHelperImpl.java]: We need here to define either the fileentry or folder as the class (based on the if condition above).

File: portal-kernel/src/com/liferay/portal/kernel/service/ServiceContextFactory.java [hunk 2]
```diff
@@ -142,7 +141,8 @@ private static void _ensureValidModelPermissions(
 		ServiceContext serviceContext) {
 
 		if (serviceContext.getModelPermissions() == null) {
-			serviceContext.setModelPermissions(new ModelPermissions());
+			serviceContext.setModelPermissions(
+				ModelPermissionsFactory.create());
```
Comment (topolik) [inline portal-kernel/src/com/liferay/portal/kernel/service/ServiceContextFactory.java]: This should be also corrected, `_ensureValidModelPermissions` should make sure it's in correct state. Having empty model is essentially wrong.

**Comment by topolik** [comment]
Thanks @yxingwu this goes the right direction I believe. 

Can you please fix the things I mentioned and resend? Thanks.

File: portal-kernel/src/com/liferay/portal/kernel/service/ServiceContextFactory.java (hunk 2 above)
Comment (yxingwu) [reply portal-kernel/src/com/liferay/portal/kernel/service/ServiceContextFactory.java]: Hi @topolik ,
I don't understand this comment. I guess you mean the original logic is not right and we need add  more code to make sure ModelPermissions in correct state.

I am trying to change the code to :
`if ((modelPermissons == null) || Validator.isNull(modelPermissons.getResourceName())) {`
`serviceContext.setModelPermissions(ModelPermissionsFactory.create());`
`}`

I am not sure my changes is right or not.

Comment (yxingwu) [reply portal-kernel/src/com/liferay/portal/kernel/service/ServiceContextFactory.java]: Hi @topolik ,
I don't understand this comment. I guess you mean the original logic is not right and we need add  more code to make sure ModelPermissions in correct state.

I am trying to change the code to :
`if ((modelPermissons == null) || Validator.isNull(modelPermissons.getResourceName())) {`
`serviceContext.setModelPermissions(ModelPermissionsFactory.create());`
`}`

I am not sure my changes is right or not.

**Pull request closed without merging**